backend/database-sqlite.js

```diff
@@ -0,0 +1,65 @@
+'use strict'
+
+import { mkdir } from 'node:fs/promises'
+import { join, resolve } from 'node:path'
+import sqlite3 from 'sqlite3'
+
+import { checkKey } from './database.js'
+
+let db: any = null
+let readStatement: any = null
+let writeStatement: any = null
+
+export async function initStorage (options: Object = {}): Promise<void> {
+  const { dirname, filename } = options
+  const dataFolder = resolve(dirname)
+
+  await mkdir(dataFolder, { mode: 0o750, recursive: true })
+
+  await new Promise((resolve, reject) => {
+    if (db) {
+      reject(new Error(`The ${filename} SQLite database is already open.`))
+    }
+    db = new sqlite3.Database(join(dataFolder, filename), (err) => {
+      if (err) {
+        reject(err)
+      }
+    })
+    db.run('CREATE TABLE IF NOT EXISTS Data(key TEXT UNIQUE NOT NULL, value TEXT NOT NULL)', [], (err) => {
```

Comment:
Also, `key` should be marked as a `PRIMARY KEY` instead of `UNIQUE`. This should also speed up performance as primary keys should be automatically indexed in sqlite. At least according to GPT4 (and my memory).